### PR TITLE
jless 0.7.1 (new formula)

### PR DESCRIPTION
**Pull request by rolandcrosby**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Adds a formula for [jless](https://pauljuliusmartinez.github.io/), a command-line JSON viewer.

### CODE REVIEW

File: Formula/jless.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "#{bin}/jless", "--help"
```
Comment (carlocab) [inline Formula/jless.rb:15]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

You should probably be able to interact with this using Ruby's PTY module.

Comment (rolandcrosby) [reply Formula/jless.rb:15]: Ah, sorry! Will figure that out and add a better test.

File: Formula/jless.rb [hunk 2]
```diff
@@ -12,6 +12,9 @@ def install
   end
 
   test do
-    system "#{bin}/jless", "--help"
+    (testpath/"example.json").write('{"hello": "world"}')
+    res, process = Open3.capture2("#{bin}/jless example.json")
+    assert_match(/"hello": "world"/, res)
```
Comment (carlocab) [inline Formula/jless.rb:17]:
```suggestion
    assert_match('"hello": "world"', res)
```
You don't need a regex here.  Alternatively:
```suggestion
    assert_equal("world", JSON.parse(res)["hello"])
```

File: Formula/jless.rb (hunk 1 above)
Comment (rolandcrosby) [reply Formula/jless.rb]: Couldn't figure out how to get the PTY module working with the interactive fullscreen thing this program does, but it also has a non-interactive mode that's easy to test by redirecting the output, so I've just tested that instead.

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by rolandcrosby** [comment]
thank you for reviewing!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @rolandcrosby for your contribution! 💯

**Pull request closed without merging**